src/index.c

```diff
@@ -915,6 +939,10 @@ static IndexCriteriaTester *NI_GetCriteriaTester(void *ctx) {
   NotContext *nc = ctx;
   NI_CriteriaTester *nct = rm_malloc(sizeof(*nct));
   nct->child = nc->base.GetCriteriaTester(nc->base.ctx);
+  if (!nct->child) {
```

Comment:
If you move line 940 and assign the ct to a temporary variable, you wouldn't need to free it if it's NULL, simplifying the code.